lib/usecases/forgot_password_use_case.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:async';
+
+import 'package:injectable/injectable.dart';
+import 'package:survey_flutter_ic/api/repository/auth_repository.dart';
+import 'package:survey_flutter_ic/model/forgot_password_model.dart';
+import 'package:survey_flutter_ic/usecases/base/base_use_case.dart';
+import 'package:survey_flutter_ic/usecases/params/forgot_password_params.dart';
+
+@Injectable()
+class ForgotPasswordUseCase
+    extends UseCase<ForgotPasswordModel, ForgotPasswordParams> {
+  final AuthRepository _repository;
+
+  const ForgotPasswordUseCase(this._repository);
+
+  @override
+  Future<Result<ForgotPasswordModel>> call(ForgotPasswordParams params) async {
+    try {
+      final result = await _repository.forgotPassword(email: params.user.email);
```

Comment:
I think you just only need to pass the email string to the function, right?

```suggestion
class ForgotPasswordUseCase
    extends UseCase<ForgotPasswordModel, String> {
  final AuthRepository _repository;

  const ForgotPasswordUseCase(this._repository);

  @override
  Future<Result<ForgotPasswordModel>> call(String params) async {
    try {
      final result = await _repository.forgotPassword(email: params);
```